src/org/thoughtcrime/securesms/ConversationFragment.java

```diff
@@ -415,11 +416,11 @@ private void handleCopyMessage(final Set<DcMsg> dcMsgsSet) {
                 result.append("\n\n");
             }
 
-            if (msg.getFromId() != prevMsg.getFromId()) {
+            if (msg.getFromId() != prevMsg.getFromId() && !singleMsg) {
                 DcContact contact = dcContext.getContact(msg.getFromId());
                 result.append(contact.getDisplayName()).append(":\n");
             }
-            if(msg.getType() == DcMsg.DC_MSG_TEXT) {
+            if(msg.getType() == DcMsg.DC_MSG_TEXT || singleMsg) {
                 result.append(msg.getText());
             } else {
                 result.append(msg.getSummarytext(10000000));
```

Comment:
that would result in copying an empty text if the user tries to copy eg. an image or a voice-message.

before, the image type was copied to the clipboard in the these cases (from summary then).

therefore, we should append the result from msg.getText() only if not empty - else use the summary.

Reply:
Don't know what the usecase of copying the word "Image" would be but anyway, done (I don't think it's an improvement but it doesn't make anything worse, either)